Sources/XCTest/XCTestRun.swift

```diff
@@ -0,0 +1,150 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//
+//  XCTestRun.swift
+//  A test run collects information about the execution of a test.
+//
+
+#if os(Linux) || os(FreeBSD)
+    import Foundation
+#else
+    import SwiftFoundation
+#endif
+
+/// A test run collects information about the execution of a test. Failures in
+/// explicit test assertions are classified as "expected", while failures from
+/// unrelated or uncaught exceptions are classified as "unexpected".
+public class XCTestRun {
+    /// The test instance provided when the test run was initialized.
+    public let test: XCTest
+
+    /// The time at which the test run was started, or nil.
+    public private(set) var startDate: NSDate?
+
+    /// The time at which the test run was stopped, or nil.
+    public private(set) var stopDate: NSDate?
+
+    /// The number of seconds that elapsed between when the run was started and
+    /// when it was stopped.
+    public var totalDuration: NSTimeInterval {
+        guard stopped else {
```

Comment:
Good call, I forgot we can unwrap multiple optionals at once now. I'll change this.